### PR TITLE
Migrate off Sapphire logger to Winston logger

### PR DESCRIPTION
**Pull request by KevinBacabac**
## Summary of Changes
Move to winston based logger.

## Motivation and Explanation
Winston is a popular library to log both plaintext and json. Using it allows us to show the log level, timestamp, stack traces. In the future we can use Winston to easily log warnings and errors to a file.

## Related Issues
https://github.com/uwcsc/codeybot/issues/316

## Steps to Reproduce
Run codey and look at the non sapphire logs in terminal.

## Demonstration of Changes
// Two examples of warn and errro
2022-08-20 19:35:31 - warn - json warn
2022-08-20 19:35:31 - error - Error
    at startBot (/usr/app/dist/src/bot.js:68:28)
    at Object.<anonymous> (/usr/app/dist/index.js:4:20)
    at Module._compile (node:internal/modules/cjs/loader:1120:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1174:10)
    at Module.load (node:internal/modules/cjs/loader:998:32)
    at Module._load (node:internal/modules/cjs/loader:839:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
// Messages from codey in this branch
2022-08-20 19:35:31 - info - {"event":"init"}
2022-08-20 19:35:32 - info - 1.0.0
        [+] Gateway
        </> DEVELOPMENT MODE
2022-08-20 19:36:00 - info - Initialized database and tables.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,8 @@
     "sqlite": "^4.0.22",
     "sqlite3": "^5.0.2",
     "stable-marriage": "^1.0.2",
-    "typescript": "^4.6.3"
+    "typescript": "^4.6.3",
+    "winston": "^3.8.1"
```
Comment (KevinBacabac) [inline package.json:44]: Add dependency

File: src/bot.ts [hunk 2]
```diff
@@ -41,13 +42,15 @@ container.botPrefix = client.options.defaultPrefix!;
 
 export const startBot = async (): Promise<void> => {
   try {
-    client.logger.info({
+    logger.info({
       event: 'init',
     });
     client.on('error', client.logger.error);
+    // Use this on the discord.js client after sapphire
+    // client.on('error', logger.error);
```
Comment (KevinBacabac) [inline src/bot.ts:50]: Sapphire requires using their logger. We can change to our logger next time

File: src/commands/coffeeChat/coffeeChat.ts [hunk 3]
```diff
@@ -55,8 +56,9 @@ export class CoffeeChatCommand extends SubCommandPluginCommand {
       } catch (err) {
         logger.error({
           event: 'client_error',
-          error: (err as Error).toString(),
+          where: 'coffeeChat alertMatches',
```
Comment (KevinBacabac) [inline src/commands/coffeeChat/coffeeChat.ts:59]: We can optionally add tags to our json

**Comment by gzcharleszhang** [comment]
we used to use winston for logging, idk what happened https://github.com/uwcsc/codeybot/pull/171/files

**Comment by mcpenguin** [comment]
probably when we migrated to sapphire

**Review by mcpenguin — COMMENTED** [review]
Seems good to me, just fix linter then should be good to merge

**Review by Picowchew — APPROVED** [review]
LGTM!

**Pull request merged**